src/c++/perf_analyzer/test_custom_load_manager.cc

```diff
@@ -33,11 +33,113 @@
+      std::uniform_int_distribution<uint64_t>& distribution)
+      : RequestRateWorker(
+            id, thread_stat, thread_config, parser, data_loader, backend_kind,
+            factory, sequence_length, start_sequence_id, sequence_id_range,
+            on_sequence_model, async, max_threads, using_json_data, streaming,
+            shared_memory_type, batch_size, sequence_stat,
+            shared_memory_regions, wake_signal, wake_mutex, execute,
+            curr_seq_id, start_time, schedule, gen_duration, distribution)
+  {
+  }
+
+  std::chrono::nanoseconds GetNextTimestamp() override
+  {
+    return RequestRateWorker::GetNextTimestamp();
+  }
+};
+
+/// Class to test the CustomLoadManager
+///
+class TestCustomLoadManager : public TestLoadManagerBase,
+                              public CustomLoadManager {
+ public:
+  TestCustomLoadManager() = default;
+
+  TestCustomLoadManager(
+      PerfAnalyzerParameters params, bool is_sequence_model = false,
+      bool is_decoupled_model = false)
+      : TestLoadManagerBase(params, is_sequence_model, is_decoupled_model),
+        CustomLoadManager(
+            params.async, params.streaming, "INTERVALS_FILE", params.batch_size,
+            params.measurement_window_ms, params.max_trials, params.max_threads,
+            params.num_of_sequences, params.sequence_length,
+            params.shared_memory_type, params.output_shm_size,
+            params.start_sequence_id, params.sequence_id_range, GetParser(),
+            GetFactory())
+  {
+    InitManager(
+        params.string_length, params.string_data, params.zero_input,
+        params.user_data);
+  }
+
+
+  std::shared_ptr<IWorker> MakeWorker(
+      std::shared_ptr<ThreadStat> thread_stat,
+      std::shared_ptr<RequestRateWorker::ThreadConfig> thread_config) override
+  {
+    uint32_t id = workers_.size();
+    return std::make_shared<TestRequestRateWorker>(
+        id, thread_stat, thread_config, LoadManager::parser_, data_loader_,
+        backend_->Kind(), RequestRateManager::factory_, sequence_length_,
+        start_sequence_id_, sequence_id_range_, on_sequence_model_, async_,
+        max_threads_, using_json_data_, streaming_, shared_memory_type_,
+        batch_size_, sequence_stat_, shared_memory_regions_, wake_signal_,
+        wake_mutex_, execute_, curr_seq_id_, start_time_, schedule_,
+        gen_duration_, distribution_);
+  }
+
+  void TestSchedule(
+      std::vector<uint64_t> intervals, PerfAnalyzerParameters params)
+  {
+    for (auto i : intervals) {
+      custom_intervals_.push_back(std::chrono::nanoseconds{i});
+    }
+    std::chrono::nanoseconds max_test_duration{params.measurement_window_ms *
+                                               1000000 * params.max_trials};
+    std::chrono::nanoseconds expected_current_timestamp{0};
+    size_t intervals_index = 0;
+
+    PauseWorkers();
+    InitCustomIntervals();
+    while (expected_current_timestamp.count() < max_test_duration.count()) {
```

Comment:
Can you add a comment to this loop explaining what it's doing?

Reply:
remove `.count()`, chromo already has less than operator defined

Reply:
added comment. removed .count for this compare (but kept it in the actual CHECK so that the values will be printed on a mismatch)